homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -189,7 +190,17 @@ def _calc_precipitation(rain, snow):
     @staticmethod
     def _get_condition(weather_code):
         """Get weather condition from weather data."""
-        return [k for k, v in CONDITION_CLASSES.items() if weather_code in v][0]
+        condition = [k for k, v in CONDITION_CLASSES.items() if 800 in v][0]
+
+        if condition == "sunny/clear-night":
```

Comment:
move `if` in the beginning of the method, we don't need to search condition if it falls into special case